### PR TITLE
Adjust pytest config

### LINKED ISSUE #3: Enable -W error in tests
I just had to disable this because I could not figure out how to get the tests to pass otherwise. But that's going to come back to bite us eventually, so we really should figure this out and fix it...

The warnings I'm seeing are:

```
tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:199: TrioDeprecationWarning: trio.hazmat.wait_socket_writable is deprecated since Trio 0.12.0; use trio.hazmat.wait_writable instead (https://github.com/python-trio/trio/issues/878)
    await trio.hazmat.wait_socket_writable(sock)

tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:190: TrioDeprecationWarning: trio.hazmat.wait_socket_readable is deprecated since Trio 0.12.0; use trio.hazmat.wait_readable instead (https://github.com/python-trio/trio/issues/878)
    await trio.hazmat.wait_socket_readable(sock)

tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:179: TrioDeprecationWarning: trio.hazmat.notify_socket_close is deprecated since Trio 0.12.0; use trio.hazmat.notify_closing instead (https://github.com/python-trio/trio/issues/878)
    trio.hazmat.notify_socket_close(self._raw_socket)

tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/uritemplate/variable.py:363: DeprecationWarning: Using or importing the ABCs tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:199: TrioDeprecationWarning: trio.hazmat.wait_socket_writable is deprecated since Trio 0.12.0; use trio.hazmat.wait_writable instead (https://github.com/python-trio/trio/issues/878)
    await trio.hazmat.wait_socket_writable(sock)

tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:190: TrioDeprecationWarning: trio.hazmat.wait_socket_readable is deprecated since Trio 0.12.0; use trio.hazmat.wait_readable instead (https://github.com/python-trio/trio/issues/878)
    await trio.hazmat.wait_socket_readable(sock)

tests/test_gh.py::test_basic_gh_client
tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/anyio/_backends/_trio.py:179: TrioDeprecationWarning: trio.hazmat.notify_socket_close is deprecated since Trio 0.12.0; use trio.hazmat.notify_closing instead (https://github.com/python-trio/trio/issues/878)
    trio.hazmat.notify_socket_close(self._raw_socket)

tests/test_gh.py::test_client_part_of_app
  /home/njs/.user-python3.7/lib/python3.7/site-packages/uritemplate/variable.py:363: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    return isinstance(value, (dict, collections.MutableMapping))
from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    return isinstance(value, (dict, collections.MutableMapping))
```

Things I tried:

First I tried `-W error:::snekomatic`, which supposedly should only turn warnings into errors if they appear in snekomatic. This actually removed the errors! But then I tried `-W error:::anyio` to make sure that the filter was working properly and... that *also* passed just fine, so apparently I don't understand how `-W` works.

Then I tried adding some `warnings.filterwarnings` calls to our `conftest.py` file. But I guess pytest applies `-W` arguments *after* loading conftest, or something? Anyway I couldn't get it to work.

If nothing else, these should both hopefully disappear on their own soon... they're both fixed upstream, just waiting for releases.

### PR DESCRIPTION
**Pull request by webknjaz**
Also includes a handy tox config

Fixes #3

### CODE REVIEW

File: ci.sh [hunk 1]
```diff
@@ -21,6 +21,6 @@ python -m pip install -r test-requirements.txt
 
 #black --diff --check snekomatic
 
-pytest --cov="snekomatic" --cov="tests" --cov-report=term -r a --verbose
+pytest $*
```
Comment (njsmith) [inline ci.sh]: I think you mean `"$@"`? Not that we're using it either way, but hey, if you don't know about `"$@"` it's a good thing to know :-)

Comment (webknjaz) [reply ci.sh]: Ah, right. Been in a hurry to show you how to solve your problem :)

**Comment by njsmith** [comment]
Thanks! And I'm amused that if things go right, I think this will be the first time snekomatic actually does its thing...

**Comment by njsmith** [comment]
Ah crud, it looks like it sent the invite but then crashed because I didn't give it permission to post comments.

**Comment by webknjaz** [comment]
:laughing: Yeah... I feel you :)

**Comment by njsmith** [comment]
@webknjaz OK don't take this the wrong way, but I'm gonna kick you out of the org so I can try redelivering the webhook and make sure the invitation works properly this time :-)

**Comment by njsmith** [comment]
What the heck. It has read-write on pull requests and issues now, but I'm still getting "gidgethub.BadRequest: Resource not accessible by integration" when it tries to post a comment.

**Comment by njsmith** [comment]
oh duh, I misread, it actually only had read-only access to pull-requests. that makes more sense then.

**Comment by njsmith** [comment]
theeeere we go

**Pull request merged**